docs: 📚 optimize code preview + add table of contents

mariohamann opened the pull request:
## Preflight Checklist

<!-- Please ensure you've completed the following steps by replacing [ ] with [x]-->

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/main/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

There's no connected issue.

## Changes

Changes in this pull request:

- This PR adds prettier formatting to every story. This was done with "Replace all" on `extractComponentDescription: () => readme,` so stories that don't use this might be missed.


## Pull request type

<!-- Please do not submit updates to dependencies unless it fixes an issue. -->

<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. -->

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

<!-- If this introduces a breaking change, please describe the impact and migration path for existing applications below. -->

## Other information

Hey there!

I'm maintainer of two design systems ([Synergy Design System](https://github.com/synergy-design-system/synergy-design-system), [Solid Design System](https://github.com/solid-design-system/solid)) that use Storybook and Lit as well. Really enjoyed going through your Storybook, but the formatting of your code really bugged me and made it feel "broken".

I therefore propose to introduce automatic formatting, which is described over here in the Storybook docs: https://storybook.js.org/docs/api/doc-block-source#format
As you already use prettier in your codebase yourself, I believe that this might be just "consequent". :)

### Example 1
#### Before
![CleanShot 2024-04-22 at 06 55 35@2x](https://github.com/lyne-design-system/lyne-components/assets/26542182/788d7978-9b36-42ed-8cc7-9e1e7cc1a4fd)

#### After
![CleanShot 2024-04-22 at 06 55 18@2x](https://github.com/lyne-design-system/lyne-components/assets/26542182/a6122556-8e04-4db1-ad1d-5d70abbc60f9)

### Example 2
#### Before
![CleanShot 2024-04-22 at 06 51 49@2x](https://github.com/lyne-design-system/lyne-components/assets/26542182/8b71be1a-3558-4f5b-92bd-49725373c72e)

#### After
![CleanShot 2024-04-22 at 06 57 06@2x](https://github.com/lyne-design-system/lyne-components/assets/26542182/c41169a4-f996-4c53-bf1a-79e8db2d5483)

All the best and keep up the cool work!

jeripeierSBB reviewed (changes requested):
Thanks a lot for your contribution. I was last week trying to integrate prettier in the browser to achieve better code block formatting. Was also checking storybook docs but just not recognizing the format source format part. So again, thanks for solving my problem! :D 
Can you please add the definition to preview.ts so we have to only once configure the formatting?

just add
```ts
  docs: {
    source: { format: 'html' },
  },
```

to `export const parameters`.

Thank you!

mariohamann commented:
Done! :) 

I wasn't sure, if you want to introduce centralized changes (we use `preview.ts` in our projects ourself for that.))

And I totally get you... we added an package for code formatting, which even introduced bugs on Safari, until I realized that there's prettier support.

Lowhangig-fruit we did in the same commit – we added Table of Contents: https://github.com/solid-design-system/solid/commit/a09daf9d50afaa5b62bac2eedc7f74b0468fb46f#diff-fa2909f2ae1c2b6b0c3710b27e556083f35005f15d2890089e8d2a431fc4ab43R89

![CleanShot 2024-04-22 at 08 50 44@2x](https://github.com/lyne-design-system/lyne-components/assets/26542182/15bbd619-7f48-4b25-b638-dd4b70f5fab4)

Do you want me to add that, too? I would update the PR title then.

jeripeierSBB commented:
If you still like to spend time for us, then go for it :D. thanks!

mariohamann commented:
![CleanShot 2024-04-22 at 09 04 25@2x](https://github.com/lyne-design-system/lyne-components/assets/26542182/783b196f-5bd0-4a45-a48b-6671f48a6a58)

That's Open Source. 😆

jeripeierSBB commented on .storybook/preview.ts:
```diff
@@ -44,6 +44,10 @@ export const parameters = {
     debounceTimeout: 10,
   },
+  docs: {
+    toc: true,
```
For now I'd prefer this config:

```suggestion
    toc: {
      ignoreSelector: '.docs-story h2',
      headingSelector: 'h2',
    },
```

jeripeierSBB reviewed (approved):
LGTM, thanks a lot for your help :)

mariohamann commented:
![CleanShot 2024-04-22 at 09 34 25@2x](https://github.com/lyne-design-system/lyne-components/assets/26542182/4cd35766-2746-405f-a514-735a860a5df7)

Seems like you have to merge. :) Enjoy!
